Assets/Scripts/UIManager.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.EventSystems;
+
+public class UIManager : MonoBehaviour
+{
+    public GameObject placement_object_prefab;
+
+    Ray mouse_ray;
+    RaycastHit hit;
+
+    GameObject tile;
+
+    GraphicRaycaster m_Raycaster;
+    PointerEventData m_PointerEventData;
+    EventSystem m_EventSystem;
+
+    public GameObject focused_placeholder;
+
+    private void Start()
```

Comment:
remove unused MonoBehaviour messages, they will get called if they are defined thus slowing execution speed even if nothing is in the method